### DISCUSSION TITLE
Can I do this with ReactPDF?

### DISCUSSION THREAD

**Discussion by Simon-44** [diegomura/react-pdf#1534] [Q&A]
Hi, I'm reading the documentation and playing a bit with the lib, but I'm not sure I can do this:
- For example the user navigates on their calendar, scrolling to a certain month
- Then they want to generate a PDF of what they are seeing right now (so here it's on me to determine what I want to generate, ie they scrolled to October 2021 so I generate a PDF of this month)
- They click on a button "Generate as a PDF", it opens a new tab displaying a typical PDF reader on which they can download and everything

And I want to do this client-side.

Is it possible? Thanks

**Comment by jeetiss** [comment]
Hi @Simon-44
Yes you can do last to steps with react-pdf, but you need dive deeper in api.

check `pdf` function, that used in all top level apis:

- create `blob` with `pdf` function
- convert `blob` to `objectUlr` with `URL.createObjectUrl`
- run `window.open(objectUlr)`, and this will open generated pdf in new tab on which user can download and everything
